pkg/cmd/cli/cmd/newapp.go

```diff
@@ -419,22 +420,33 @@ func setAppConfigLabels(c *cobra.Command, config *newcmd.AppConfig) error {
 	return nil
 }
 
-func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args []string, config *newcmd.AppConfig) error {
-	namespace, _, err := f.DefaultNamespace()
-	if err != nil {
-		return err
-	}
-
+// getDockerClient returns a client capable of communicating with the local
+// docker daemon.  If an error occurs (such as no local daemon being available),
+// it will return nil.
+func getDockerClient() (*docker.Client, error) {
 	dockerClient, _, err := dockerutil.NewHelper().GetClient()
 	if err == nil {
-		if err = dockerClient.Ping(); err == nil {
-			config.SetDockerClient(dockerClient)
-		} else {
+		if err = dockerClient.Ping(); err != nil {
 			glog.V(4).Infof("Docker client did not respond to a ping: %v", err)
+			return nil, err
 		}
+		return dockerClient, nil
 	}
+	glog.V(2).Infof("No local Docker daemon detected: %v", err)
+	return nil, err
+}
+
+func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args []string, config *newcmd.AppConfig) error {
+	namespace, _, err := f.DefaultNamespace()
 	if err != nil {
-		glog.V(2).Infof("No local Docker daemon detected: %v", err)
+		return err
+	}
+
+	dockerClient, err := getDockerClient()
+	if err == nil && dockerClient != nil {
+		// TODO: why do we even do this, since we immediately override
+		// c.dockerSearcher with the call to config.SetOpenShiftClient below?
+		config.SetDockerClient(dockerClient)
```

Comment:
@smarterclayton bump.  i'm inclined to delete this whole block since we already always call config.SetOpenShiftClient, but i'm worried i'm missing something about intent here.